core/src/main/java/org/hibernate/ogm/datastore/document/association/spi/AssociationRows.java

```diff
@@ -28,7 +27,7 @@
 	private final Map<RowKey, AssociationRow<?>> rows;
 
 	public AssociationRows(AssociationKey associationKey, Collection<?> wrapped, AssociationRowFactory associationRowFactory) {
-		this.rows = newHashMap( wrapped.size() );
+		this.rows = new LinkedHashMap<>( wrapped.size() );
```

Comment:
Please add a new `newLinkedHashMap` method to our CollectionHelper.

I'll work on pushing the changes we made in HV regarding the sizing.

(With the current code, you always undersize the map)